shell/platform/darwin/ios/framework/Source/accessibility_bridge.mm

```diff
@@ -431,11 +460,42 @@ - (BOOL)accessibilityScroll:(UIAccessibilityScrollDirection)direction {
   platform_view_->DispatchSemanticsAction(uid, action, args);
 }
 
-SemanticsObject* AccessibilityBridge::GetOrCreateObject(int32_t uid) {
+SemanticsObject* AccessibilityBridge::GetOrCreateObject(int32_t uid,
+                                                        blink::SemanticsNodeUpdates& updates) {
   SemanticsObject* object = objects_.get()[@(uid)];
   if (!object) {
-    object = [[[SemanticsObject alloc] initWithBridge:this uid:uid] autorelease];
+    // New node case: simply create a new SemanticsObject.
+    blink::SemanticsNode node = updates[uid];
+    if (node.HasFlag(blink::SemanticsFlags::kIsTextField)) {
+      // Text fields are backed by objects that implement UITextInput.
+      object = [[[TextInputSemanticsObject alloc] initWithBridge:this uid:uid] autorelease];
+    } else {
+      object = [[[FlutterSemanticsObject alloc] initWithBridge:this uid:uid] autorelease];
+    }
+
     objects_.get()[@(uid)] = object;
+  } else {
+    // Existing node case
+    auto nodeEntry = updates.find(object.node.id);
+    if (nodeEntry != updates.end()) {
+      // There's an update for this node
+      blink::SemanticsNode node = nodeEntry->second;
+      BOOL isTextField = node.HasFlag(blink::SemanticsFlags::kIsTextField);
+      BOOL wasTextField = object.node.HasFlag(blink::SemanticsFlags::kIsTextField);
+      if (wasTextField != isTextField) {
```

Comment:
I believe, the way this is set up we will only act correctly on a switch from textfield to non-textfield (and vice versa) if the parent of the switched node is included in the semantics update? If only the switched node is included in the semantics update, there seems to be nothing updating that node's parent's child list to include the new node?

Maybe, instead of holding on to child objects, we need to switch that to just ids? And whenever you need the real object you need to look it up in the objects_ map, which would be the ultimate truth?

Reply:
You're right. I need to restructure this. OTOH, why _don't_ we include parents in the update though? Tree depth is O(log(N)) of the size of the tree, so omitting parent nodes doesn't save anything performance-wise. If we had parents with the update, we could update everything in a single depth-first pass. Also, it would remove the need for the `objects_` map.

WDYT?

Reply:
The ideal case is that we have updates that send a single node only, so sending more nodes would be less good. :-)

Reply:
Right now, we assume that every node included in the update has changed (and on Android we need to tell the OS about every node that has changed). If we now include unchanged nodes, we'd have to mark those in the update to avoid reporting unchanged nodes as changed...

Reply:
I was thinking of adding a `SemanticsNodeUpdate` class that _has_ a `SemanticsNode` containing updated data. `nullptr` would indicate "there no update to the node itself, only its children".

However, that's coming way too close to @chinmaygarde's proposal for a unified retained-mode semantics tree, so I'm not going to do this as part of this change, and just swap out the child reference using the current system. We can come back to this later.

Reply:
see https://github.com/flutter/engine/pull/4575/commits/4fc35accddf589d1e4636255994f123a4aa2494a

Reply:
update, after rebase it's now https://github.com/flutter/engine/pull/4575/commits/5baa9279e5c2a6fe174cee9687c608d88554beac